x/mongo/driver/operation.go

```diff
@@ -1540,34 +1601,29 @@ func (op Operation) addClusterTime(dst []byte, desc description.SelectedServer)
 // operation's MaxTimeMS if set. If no MaxTimeMS is set on the operation, and context is
 // not a Timeout context, calculateMaxTimeMS returns 0.
 func (op Operation) calculateMaxTimeMS(ctx context.Context, rttMin time.Duration, rttStats string) (uint64, error) {
-	if csot.IsTimeoutContext(ctx) {
-		if deadline, ok := ctx.Deadline(); ok {
-			remainingTimeout := time.Until(deadline)
-
-			// Always round up to the next millisecond value so we never truncate the calculated
-			// maxTimeMS value (e.g. 400 microseconds evaluates to 1ms, not 0ms).
-			maxTimeMS := int64((remainingTimeout - rttMin + time.Millisecond - 1) / time.Millisecond)
-			if maxTimeMS <= 0 {
-				return 0, fmt.Errorf(
-					"remaining time %v until context deadline is less than or equal to rtt minimum: %w\n%v",
-					remainingTimeout,
-					ErrDeadlineWouldBeExceeded,
-					rttStats)
-			}
+	if csot.IsWithoutMaxTime(ctx) {
+		return 0, nil
+	}
```

Comment:
We should try to avoid passing operation data like the "omit maxTimeMS" flag through a Context. From [Context best practices](https://go.dev/wiki/CodeReviewComments#contexts):
> If you have application data to pass around, put it in a parameter, in the receiver, in globals, or, if it truly belongs there, in a Context value.

Is there another way we can communicate the "omit maxTimeMS" flag to `Operation.Execute`?

Reply:
I removed this constructor in favor of a new driver.Operation field, `OmitMaxtTimeMS`.